src/components/Send.jsx

```diff
@@ -1,17 +1,103 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { useState } from 'react'
 import { useLocation } from 'react-router-dom'
 import * as rb from 'react-bootstrap'
-import { serialize, ACCOUNTS } from '../utils'
+import PageTitle from './PageTitle'
+import ToggleSwitch from './ToggleSwitch'
+import { useCurrentWalletInfo, useSetCurrentWalletInfo, useCurrentWallet } from '../context/WalletContext'
+import { useSettings } from '../context/SettingsContext'
+import { serialize } from '../utils'
 import * as Api from '../libs/JmWalletApi'
 
-export default function Payment({ currentWallet }) {
+const CollaboratorsSelector = ({ numCollaborators, setNumCollaborators }) => {
+  const [usesCustomNumCollaborators, setUsesCustomNumCollaborators] = useState(false)
+
+  const collaboratorsSelection = [3, 5, 6, 9]
+
+  return (
+    <rb.Form.Group className="collaborators-selector">
+      <rb.Form.Label className="mb-0">Choose a number of collaborators ({numCollaborators})</rb.Form.Label>
+      <div className="mb-2">
+        <rb.Form.Text className="text-secondary">
+          A higher number is better for privacy, but also increases the fee.
+        </rb.Form.Text>
+      </div>
+      <div className="d-flex flex-row flex-wrap">
+        {collaboratorsSelection.map((number, index) => {
+          return (
+            <rb.Button
+              key={index}
+              variant="white"
+              className={`py-2 px-2 border rounded text-center${
+                !usesCustomNumCollaborators && numCollaborators === number ? ' border-dark border-2' : ''
+              }`}
+              onClick={() => {
+                setUsesCustomNumCollaborators(false)
+                setNumCollaborators(number)
+              }}
+            >
+              {number}
+            </rb.Button>
+          )
+        })}
+        <rb.Form.Control
+          type="number"
+          min={1}
+          placeholder="Other"
+          className={`py-2 px-2 border rounded text-center${usesCustomNumCollaborators ? ' border-dark border-2' : ''}`}
+          onChange={(e) => {
+            setUsesCustomNumCollaborators(true)
+            setNumCollaborators(e.target.value)
+          }}
+          onClick={(e) => {
+            if (e.target.value) {
+              setUsesCustomNumCollaborators(true)
+              setNumCollaborators(e.target.value)
+            }
+          }}
+        />
+      </div>
+      <rb.Form.Control.Feedback type="invalid">Please set the counterparties.</rb.Form.Control.Feedback>
```

Comment:
```suggestion
      {usesCustomNumCollaborators && <rb.Form.Control.Feedback type="invalid">Please set the counterparties.</rb.Form.Control.Feedback>}
      </div>
```
I have not managed to get this message to be displayed otherwise.

However:
- If you provide an invalid value (e.g. `-1`), the default value is shown as "selected", but the form cannot be sent without visual feedback

Reply:
Updated. ✅ 

> If you provide an invalid value (e.g. -1), the default value is shown as "selected", but the form cannot be sent without visual feedback

Can you elaborate on that. Could not reproduce it. I feel like I might be missing something.

Reply:
> Updated. white_check_mark
:+1:
> Can you elaborate on that. Could not reproduce it. I feel like I might be missing something.

My wording was not clear - here are other steps to also reproduce it:
- insert a 'abc' in the "other" input -> feedback is shown now
- select a number e.g. 6 (but leave 'a'!) -> feedback disappears
- try sending the form -> form cannot be sent

Reply:
Oh I get it now! Yeah you're right that whole part needs a little work still. Will update this tomorrow. 😄